packages/app/src/pages/dashboard/endpoints/_/body/item/edit/index.tsx

```diff
@@ -0,0 +1,164 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import React, { useCallback, useMemo } from 'react';
+import { useForm } from 'react-hook-form';
+import * as yup from 'yup';
+import { SIZE as BUTTON_SIZE } from '~/components/button';
+import Button, { Props as ButtonProps } from '~/components/button';
+import Head from '~/components/head';
+import Select from '~/components/select';
+import Textinput from '~/components/textinput';
+import { useEndpoint } from '~/hooks/endpoint';
+import { useTranslation } from '~/hooks/i18n';
+import { ClassName, COLOR_SYSTEM, Endpoint, EndpointGroup } from '~/types';
+import { endpointId, url } from '~/utils/v8n';
+
+export type Props = {
+  className?: ClassName;
+  onAdd: () => void;
+  onCancel: () => void;
+  endpoint: Endpoint;
+};
+
+const EditEndpoint: React.FC<Props> = ({
+  className = '',
```

Comment:
imo: If this is the standard practice in this project for managing fallbacks on optional props, then it's acceptable. However, if it's not, it might be redundant since it doesn't affect anything.